### PR TITLE
qa: Update tests and remove unused `prophecy` dependency

### PR DESCRIPTION
**Pull request by alexraputa**
|    Q          |   A
|-------------- | ------
| Documentation | no
| Bugfix        | no
| BC Break      | no
| New Feature   | no
| RFC           | no
| QA            | yes

### CODE REVIEW

File: test/Middleware/ImplicitHeadMiddlewareTest.php [hunk 1]
```diff
@@ -48,11 +48,13 @@ public function testReturnsResultOfHandlerOnNonHeadRequests(): void
     {
         $request = $this->createMock(ServerRequestInterface::class);
         $request
+            ->expects(self::once())
```
Comment (Ocramius) [inline test/Middleware/ImplicitHeadMiddlewareTest.php]: We don't really care about `once()` on reads on stubs - careful where you add it

Comment (Ocramius) [reply test/Middleware/ImplicitHeadMiddlewareTest.php]: This applies across the whole patch: careful where you add `once()` or `exactly()`, as it is an assertion

**Review by Ocramius — CHANGES_REQUESTED** [review]

File: test/Middleware/ImplicitHeadMiddlewareTest.php (hunk 1 above)
Comment (alexraputa) [reply test/Middleware/ImplicitHeadMiddlewareTest.php]: revert?

Comment (Ocramius) [reply test/Middleware/ImplicitHeadMiddlewareTest.php]: IMO to be fixed up with a new commit introducing clarity there

Comment (alexraputa) [reply test/Middleware/ImplicitHeadMiddlewareTest.php]: @Ocramius, done.

**Review by Ocramius — APPROVED** [review]
Thanks @alexraputa!

**Pull request merged**